### PR TITLE
New Resource: aws_waf_rule_group

### PR DESCRIPTION
**Pull request by radeksimko**
## Test results

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSWafRuleGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSWafRuleGroup_ -timeout 120m
=== RUN   TestAccAWSWafRuleGroup_basic
--- PASS: TestAccAWSWafRuleGroup_basic (36.98s)
=== RUN   TestAccAWSWafRuleGroup_changeNameForceNew
--- PASS: TestAccAWSWafRuleGroup_changeNameForceNew (63.07s)
=== RUN   TestAccAWSWafRuleGroup_disappears
--- PASS: TestAccAWSWafRuleGroup_disappears (34.66s)
=== RUN   TestAccAWSWafRuleGroup_changeActivatedRules
--- PASS: TestAccAWSWafRuleGroup_changeActivatedRules (67.41s)
=== RUN   TestAccAWSWafRuleGroup_noActivatedRules
--- PASS: TestAccAWSWafRuleGroup_noActivatedRules (26.29s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	228.450s
```

### CODE REVIEW

File: aws/resource_aws_waf_rule_group.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+		return conn.UpdateRuleGroup(req)
+	})
+	if err != nil {
+		return fmt.Errorf("Error Updating WAF Rule Group: %s", err)
+	}
+
+	return nil
+}
+
+func diffWafRuleGroupActivatedRules(oldRules, newRules []interface{}) []*waf.RuleGroupUpdate {
```
Comment (appilon) [inline aws/resource_aws_waf_rule_group.go]: could this differ be transplanted to `waf_helpers.go`. Does WafRegional have the equivalent resource

File: aws/resource_aws_waf_rule_group.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+		updates = append(updates, &waf.RuleGroupUpdate{
+			Action:        aws.String(waf.ChangeActionInsert),
+			ActivatedRule: expandWafActivatedRule(rule),
+		})
+	}
+	return updates
+}
+
+func flattenWafActivatedRules(activatedRules []*waf.ActivatedRule) []interface{} {
```
Comment (appilon) [inline aws/resource_aws_waf_rule_group.go]: Again could it be moved to helper file?

File: aws/resource_aws_waf_rule_group.go [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+					"type": *ar.Action.Type,
+				},
+			}
+		}
+		out[i] = rule
+	}
+	return out
+}
+
+func expandWafActivatedRule(rule map[string]interface{}) *waf.ActivatedRule {
```
Comment (appilon) [inline aws/resource_aws_waf_rule_group.go]: could be moved to helper?

File: aws/resource_aws_waf_rule_group_test.go [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+	var before, after waf.RuleGroup
+
+	ruleName := fmt.Sprintf("tfacc%s", acctest.RandString(5))
+	groupName := fmt.Sprintf("tfacc%s", acctest.RandString(5))
+	newGroupName := fmt.Sprintf("tfacc%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafIPSetDestroy,
```
Comment (appilon) [inline aws/resource_aws_waf_rule_group_test.go]: wrong destroy check?

**Comment by radeksimko** [comment]
@appilon Thanks for the review, I believe I addressed all your feedback, PTAL.

**Comment by bflad** [comment]
This has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**